### DISCUSSION TITLE
How to pass Option<vector<u8>> using ts-sdk?

### DISCUSSION THREAD

**Discussion by ajaythxkur** [aptos-labs/aptos-developer-discussions#444] [Questions] [answered]
` const serializer = new Serializer();

  serializer.serializeStr(coin.asset_type);

  coin_type = serializer.toUint8Array();

  const deserializer = new Deserializer(serializer.toUint8Array());

  console.log({deserialized: deserializer.deserializeStr()})
`
Passing 
`0xc21eef93e0188165bc9f303e7f8b7f24064db5e6981d1cd092ee4a4b84ac38af::coin::MyCoin`

resulting in 

`Y0xc21eef93e0188165bc9f303e7f8b7f24064db5e6981d1cd092ee4a4b84ac38af::coin::MyCoin`

**Comment by gregnazario** [comment, marked as answer]
## Encoding an `Option<vector<u8>>`
### As an entry function argument:

If it's going into an entry function, you can just provide the string, and the ABI will figure it out as a filled or not filled option.  `undefined` or `null` represent `Option::none()` and any value (in this case `Uint8array`) represents `Option::some()`

The most accurate way, you can just serialize like below, and pass in optionVec as an input argument:
```ts
    const vecU8 = MoveVector.U8(input)
    const optionVec = new MoveOption(vecU8);
```

## For BCS Encoding:

```ts
import {
    Deserializer,
    Hex,
    HexInput, MoveOption, MoveVector,
    Serializable,
    Serializer, U8
} from "@aptos-labs/ts-sdk";
function deserializeOptionVec(deserializer: Deserializer): MoveOption<MoveVector<U8>> {
    const bool = deserializer.deserializeBool();
    if (!bool) {
        return new MoveOption(null);
    }
    const vec = MoveVector.deserialize(deserializer, U8);
    return new MoveOption(vec);
}

const example = async () => {
    // Serializing
    const input = "0x012345";
    const serializer = new Serializer();
    const vecU8 = MoveVector.U8(input)
    console.log(`Input: ${input}`);
    const optionVec = new MoveOption(vecU8);
    serializer.serialize(optionVec);
    const bytes = serializer.toUint8Array();
    console.log(`Encoded: ${Hex.fromHexInput(bytes).toString()}`);

    // Deserializing
    const deserializer = new Deserializer(bytes);
    const deserializedOptionVec = deserializeOptionVec(deserializer);
    const deserializedVec = new Uint8Array(deserializedOptionVec.value.values.map((u8) => u8.value));
    console.log(`Decoded: ${Hex.fromHexInput(deserializedVec).toString()}`);
}
example();
```

**Reply by ajaythxkur** [reply]
Thanks, passing the string directly worked!! @gregnazario thanks for saving my day
